app/src/main/java/com/github/se/travelpouch/model/TravelContainer.kt

```diff
@@ -0,0 +1,93 @@
+package com.github.se.travelpouch.model
+
+import com.google.firebase.Timestamp
+
+/**
+ * Data class representing a travel container.
+ *
+ * @property fsUid Firestore UID of the travel container.
+ * @property title Title of the travel.
+ * @property description Description of the travel container, can be a blank string.
+ * @property startTime Start time of the travel.
+ * @property endTime End time of the travel.
+ * @property location Location details of the travel.
+ * @property allAttachments Map of attachments, name of the attachment and Firestore UID of the
+ *   attachment.
+ * @property allParticipants Map of participants, Participant and their Role. One owner must exist
+ *   at all times unless the travel container is deleted.
+ */
+data class TravelContainer(
+    val fsUid: String, // Firestore UID
+    val title: String,
+    val description: String, // can be a blank string
+    val startTime: Timestamp,
+    val endTime: Timestamp,
+    val location: Location,
+    val allAttachments:
+        Map<
+            String,
+            String>, // list of attachments, key is the name of the attachment, value is the uid of
+    // the attachment
+    val allParticipants:
+        Map<
+            Participant,
+            Role>, // One owner must exist at all times unless travel container is deleted
+) {
+  init {
+    require(allParticipants.isNotEmpty()) { "At least one participant is required" }
+    require(allParticipants.values.contains(Role.OWNER)) { "At least one owner is required" }
+  }
```

Comment:
We can add some verifications to ensure the robustness of the code:

- _Title_ attribute cannot be empty because if the title serves as a reference for displaying and distinguishing TravelContainer instances, allowing it to be empty might not be ideal for the following reasons: 
  - **User Experience**: An empty title could lead to confusion when users are trying to differentiate between various travel containers. Having a meaningful title improves the clarity of the UI; 
  - **Data Integrity**: By enforcing a non-blank title, you ensure that every TravelContainer has a valid identifier, reducing the chance of displaying "Untitled" or similar placeholders in your application; 
  - **Consistency**: Consistent data quality is essential for maintaining a reliable data model. Requiring a non-empty title can contribute to more predictable behavior in your application.
- It is essential to ensure that the startTime is always before the endTime. This validation can prevent logical errors and improve data integrity. 
`require(startTime.toDate().before(endTime.toDate())) { "startTime must be before endTime" }`
- To ensure that the _fsUid_ is valid and not blank, you should add a validation check in the initializer. This will help catch errors early. See also **data class Participant** for checking _fsUid_
`require(fsUid.isNotBlank() && isValidUid(fsUid)) { "Invalid fsUid format" }`


```suggestion
      require(title.isNotBlank()) { "Title cannot be blank" }
      require(startTime.toDate().before(endTime.toDate())) { "startTime must be before endTime" }
      require(fsUid.isNotBlank() && isValidUid(fsUid)) { "Invalid fsUid format" }
  }
```